libknet/handle.c

```diff
@@ -706,6 +731,7 @@ int knet_handle_free(knet_handle_t knet_h)
 
 	pthread_rwlock_unlock(&knet_h->global_rwlock);
 
+	_stop_transports(knet_h);
```

Comment:
On shutdown, we probably want to stop the RX/TX threads before freeing sockets and transport stuff underneath.